spyderlib/otherplugins.py

```diff
@@ -8,52 +8,140 @@
 Spyder third-party plugins configuration management
 """
 
-import os
-import os.path as osp
+import importlib
 import sys
 import traceback
 
 # Local imports
-from spyderlib.utils import programs
-
-
-# Calculate path to `spyderplugins` package, where Spyder looks for all 3rd
-# party plugin modules
-PLUGIN_PATH = None
-if programs.is_module_installed("spyderplugins"):
-    import spyderplugins
-    PLUGIN_PATH = osp.abspath(spyderplugins.__path__[0])
-    if not osp.isdir(PLUGIN_PATH):
-        # py2exe/cx_Freeze distribution: ignoring extra plugins
-        PLUGIN_PATH = None
-
-
-def get_spyderplugins(prefix, extension):
-    """Scan directory of `spyderplugins` package and
-    return the list of module names matching *prefix* and *extension*"""
-    plist = []
-    if PLUGIN_PATH is not None:
-        for name in os.listdir(PLUGIN_PATH):
-            modname, ext = osp.splitext(name)
-            if prefix is not None and not name.startswith(prefix):
-                continue
-            if extension is not None and ext != extension:
-                continue
-            plist.append(modname)
-    return plist
-
-
-def get_spyderplugins_mods(prefix, extension):
-    """Import modules that match *prefix* and *extension* from
-    `spyderplugins` package and return the list"""
-    modlist = []
-    for modname in get_spyderplugins(prefix, extension):
-        name = 'spyderplugins.%s' % modname
-        try:
-            __import__(name)
-            modlist.append(sys.modules[name])
-        except Exception:
-            sys.stderr.write(
-                "ERROR: 3rd party plugin import failed for `%s`\n" % modname)
-            traceback.print_exc(file=sys.stderr)
+from spyderlib.config.base import get_conf_path
+from spyderlib.py3compat import PY2, PY3, PY33
+from spyderlib.utils.external.path import Path
+
+if PY2:
+    import imp
+
+
+def _get_spyderplugins(plugin_path, base_namespace, plugins_namespace,
+                       modnames, modlist):
+    """Scan the directory `plugin_path` for plugins_namespace package and
+    loads its submodules."""
+    namespace_path = Path(plugin_path) / base_namespace / plugins_namespace
+
+    if not namespace_path.exists():
+        return
+
+    for dirname in namespace_path.dirs():
+        if dirname.name == "__pycache__":
+            continue
+        _import_plugin(dirname.name, base_namespace, plugins_namespace,
+                       namespace_path, modnames, modlist)
+
+    for name in namespace_path.files(pattern="*.py"):
+        if name.name == "__init__.py":
+            continue
+        _import_plugin(name.namebase, base_namespace, plugins_namespace,
+                       namespace_path, modnames, modlist)
+
+
+class _ModuleMock():
+    """This mock module is added to sys.modules on plugin load to add the
+    location of the LOCALEDATA so that the module loads succesfully.
+    Once loaded the module is replaced by the actual loaded module object.
+    """

```

Comment:
Can't we add a `pass` here? Just to put some content on it, and also show that does nothing :-)